### ISSUE TITLE
Rule proposal: `prefer-array-find-last`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1323]
Prefer `.findLast()` over `.filter().at(-1)`, since `findLast` breaks the loop as soon as it finds a match.

## Fail

```js
const item = array.filter(x => x === '🦄').at(-1);
```

## Pass

```js
const item = array.findLast(x => x === '🦄');
```

Modified from [`prefer-array-find` proposal](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/730)

----

Additional checks:

```js
const item = array.filter(x => x === '🦄').pop();
```

```js
const item = [...array].reverse().find(x => x === '🦄');
```

---

The `Array#findLast()` proposal  is still at  stage 2, we need to wait until it became more stable.

- [Proposal for .findLast() and .findLastIndex() methods on array and typed array.](https://github.com/tc39/proposal-array-find-from-last)

**Comment by sindresorhus** [comment]
Not sure if anyone would actually do this, but another fail:

```js
const item = array.filter(x => x === '🦄').slice(0, -1)
```

**Comment by fisker** [comment]
You case returns an array, not the last matched element.

**Comment by fisker** [comment]
Still not right?

```js
const item = array.filter(x => x === '🦄').slice(-1)[0]
```

**Comment by sindresorhus** [comment]
Agh. I need to get some coffee...

**Comment by sindresorhus** [comment]
Accepted.

**Comment by fisker** [comment]
"Array find from last" proposal has reached stage 3 https://github.com/tc39/proposals/commit/4f68a10cad75a30cbb19fa625d4b588422908216
